### ISSUE TITLE
Yolov8 loads other datasets

### ISSUE THREAD

**Issue by Vayne0227** [ultralytics/ultralytics#14180]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello author, I would like to use YOLOv8 for multimodal object detection. But I don't know how to load my own dataset because its file structure is different from the dataset used in YOLOv8. The structure is as follows. Can you guide me on how to read this dataset?
![bc70add7315d15d595804a0c4693cc3](https://github.com/ultralytics/ultralytics/assets/66624169/082c9572-2f5f-484e-9dc4-5167df76af48)

**Comment by glenn-jocher** [comment]
@Vayne0227 hi there! 👋

Thank you for reaching out with your question. To use YOLOv8 with your custom dataset, you'll need to ensure that your dataset is formatted correctly for YOLOv8. The Ultralytics YOLO format is quite flexible and supports various dataset structures. Here's a step-by-step guide to help you get started:

1. **Dataset Structure**: Your dataset should be organized with separate directories for images and labels. Each image should have a corresponding label file with the same name but a `.txt` extension. The labels should be in the format `class x_center y_center width height`, with coordinates normalized between 0 and 1.

2. **Create a Dataset Configuration File**: You'll need a `.yaml` file to define the paths to your training, validation, and test datasets, as well as the class names. Here’s an example configuration:

    ```yaml
    path: ../datasets/your_dataset  # dataset root dir
    train: images/train  # train images (relative to 'path')
    val: images/val  # val images (relative to 'path')
    test: images/test  # test images (optional)

    # Classes
    names:
      0: class_name_1
      1: class_name_2
      # Add more classes as needed
    ```

3. **Convert Annotations**: If your annotations are not in the YOLO format, you’ll need to convert them. For example, if you have COCO format annotations, you can use the following code snippet to convert them:

    ```python
    from ultralytics.data.converter import convert_coco

    convert_coco(labels_dir="path/to/coco/annotations/")
    ```

4. **Training the Model**: Once your dataset is properly formatted and the configuration file is ready, you can train your model using the following command:

    ```python
    from ultralytics import YOLO

    # Load a model
    model = YOLO("yolov8n.pt")  # load a pretrained model (recommended for training)

    # Train the model
    results = model.train(data="path/to/your_dataset.yaml", epochs=100, imgsz=640)
    ```

For more detailed information on dataset formatting and training, you can refer to the [Ultralytics documentation](https://docs.ultralytics.com/datasets/detect/).

If you encounter any issues or need further assistance, please provide a reproducible example of your dataset structure and any error messages you receive. This will help us diagnose the problem more effectively. You can find more information on creating a minimum reproducible example [here](https://docs.ultralytics.com/help/minimum_reproducible_example).

Lastly, please ensure you are using the latest version of the Ultralytics packages to avoid any issues that may have already been resolved in recent updates.

Happy training! 🚀
